code/game/objects/items/RCD.dm

```diff
@@ -276,6 +402,123 @@ GLOBAL_VAR_INIT(icon_holographic_window, init_holographic_window())
+			"North" = list("icon" = icon_list["cnorth"], "coordinates" = list("Root" = "Construction", "Category" = "Machines", "Index" = 2)),
+			"East" = list("icon" = icon_list["ceast"], "coordinates" = list("Root" = "Construction", "Category" = "Machines", "Index" = 4)),
+			"South" = list("icon" = icon_list["csouth"], "coordinates" = list("Root" = "Construction", "Category" = "Machines", "Index" = 3)),
+			"West" = list("icon" = icon_list["cwest"], "coordinates" = list("Root" = "Construction", "Category" = "Machines", "Index" = 5)),
+		),
+		"Furnishing" = list(
+			"Chair" = list("icon" = icon_list["chair"], "coordinates" = list("Root" = "Construction", "Category" = "Furniture", "Index" = 1)),
+			"Stool" = list("icon" = icon_list["stool"], "coordinates" = list("Root" = "Construction", "Category" = "Furniture", "Index" = 2)),
+			"Table" = list("icon" = icon_list["table"], "coordinates" = list("Root" = "Construction", "Category" = "Furniture", "Index" = 3)),
+			"Glass Table" = list("icon" = icon_list["glass_table"], "coordinates" = list("Root" = "Construction", "Category" = "Furniture", "Index" = 4)),
+		),
+	)
+
+	//initialize airlock radial menus
+	var/list/airlocks = list(
+		"Standard" = 'icons/obj/doors/airlocks/station/public.dmi',
+		"Public" = 'icons/obj/doors/airlocks/station2/glass.dmi',
+		"Engineering" = 'icons/obj/doors/airlocks/station/engineering.dmi',
+		"Atmospherics" = 'icons/obj/doors/airlocks/station/atmos.dmi',
+		"Security" = 'icons/obj/doors/airlocks/station/security.dmi',
+		"Command" = 'icons/obj/doors/airlocks/station/command.dmi',
+		"Medical" = 'icons/obj/doors/airlocks/station/medical.dmi',
+		"Research" = 'icons/obj/doors/airlocks/station/research.dmi',
+		"Freezer" = 'icons/obj/doors/airlocks/station/freezer.dmi',
+		"Virology" = 'icons/obj/doors/airlocks/station/virology.dmi',
+		"Mining" = 'icons/obj/doors/airlocks/station/mining.dmi',
+		"Maintenance" = 'icons/obj/doors/airlocks/station/maintenance.dmi',
+		"External" = 'icons/obj/doors/airlocks/external/external.dmi',
+		"External-Maintenance" = 'icons/obj/doors/airlocks/station/maintenanceexternal.dmi',
+		"Airtight-Hatch" = 'icons/obj/doors/airlocks/hatch/centcom.dmi',
+		"Maintenance-Hatch" = 'icons/obj/doors/airlocks/hatch/maintenance.dmi'
+	)
+	var/list/exclusion = list("Freezer", "Airtight-Hatch", "Maintenance Hatch")
+	radial_menu["Solid"] = list()
+	radial_menu["Glass"] = list()
+
+	var/skip
+	var/solid_ref=1
+	var/glass_ref=1
+	for(var/airlock_name as anything in airlocks)
+		skip = FALSE
+
+		icon = icon(icon = airlocks[airlock_name] , icon_state = "closed" , dir = SOUTH)
+		icon.Blend(icon(icon = airlocks[airlock_name], icon_state = "fill_closed", dir = SOUTH), ICON_OVERLAY)
+		radial_menu["Solid"][airlock_name] = list("icon" = icon, "coordinates" = list("Root" = "AirLocks", "Category" = "Solid AirLocks", "Index" = solid_ref++))
+
+		for(var/exclude as anything in exclusion)
+			if(airlock_name == exclude)
+				skip = TRUE
+				break
```

Comment:
Replace with `airlock_name in exclude`.

Reply:
Yup . Code part of radial menu waiting for your word to scrap all of this